src/components/structures/MatrixChat.js

```diff
@@ -851,7 +887,12 @@ module.exports = React.createClass({
      */
     _onLoadCompleted: function() {
         this.props.onLoadCompleted();
-        this.setState({loading: false});
+
+        // if we've got this far without leaving the 'loading' view, then
+        // login must have failed, so start the login process
+        if (this.state.view === VIEWS.LOADING) {
```

Comment:
ugh. well, we don't hit that catch unless something very much blows up. The thing I *could* do is return a result from Lifecycle.loadSession and stick it in the params of the `on_load_completed` dispatch... but it's a bit faffy to thread all the way through.